public/index.html

```diff
@@ -144,6 +144,13 @@
 
 
 <div id="contentWrapper" class="container-fluid">
+
+
+    <div id="loadingWheel">
```

Comment:
I think you're going to discover that planting a giant spinner wheel in the middle of the UI everytime you touch storage is a bit too invasive.